compiler/circle-eval-diff/src/CircleEvalDiff.cpp

```diff
@@ -27,6 +27,22 @@
 namespace
 {
 
+template <typename T> bool same_shape(const T a, const T b)
```

Comment:
Can I ask you why template is needed?
With this PR, i think all of the parameters are `luci::CircleNode` :D

Reply:
I just copied this code from another file. It is not necessary to be a template. Thanks for finding.

I've updated the code.